google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -43,12 +47,32 @@
       new Timestamp(253402300799L, (int) TimeUnit.SECONDS.toNanos(1) - 1);
 
   /** Regexp to split timestamps into date-hour-minute-second and fractional second components. */
-  private static final Pattern FORMAT_REGEXP = Pattern.compile("([^\\.]*)(\\.\\d{0,9})?Z");
+  private static final Pattern FORMAT_REGEXP = Pattern.compile("([^.]*)(\\.\\d{0,9})?Z");
+  private static final String DATE_GROUP = "(\\d{4}-\\d{1,2}-\\d{1,2})";
```

Comment:
According to the BigQuery docs (and Spanner docs), this doesn't appear to be RFC 3339:

https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#canonical-format_3